sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuParquetScan.scala

```diff
@@ -924,13 +924,17 @@ class MultiFileParquetPartitionReader(
   private def addPartitionValues(
       batch: Option[ColumnarBatch],
       inPartitionValues: InternalRow): Option[ColumnarBatch] = {
-    batch.map { cb =>
-      val partitionValues = inPartitionValues.toSeq(partitionSchema)
-      val partitionScalars = ColumnarPartitionReaderWithPartitionValues
-        .createPartitionValues(partitionValues, partitionSchema)
-      withResource(partitionScalars) { scalars =>
-        ColumnarPartitionReaderWithPartitionValues.addPartitionValues(cb, scalars)
+    if (partitionSchema.nonEmpty) {
```

Comment:
Technically this change is not needed I originally started it to only grab the semaphore when there were partition values, but then I thought better and decided to grab it in all cases.  I left this because I thought it made the code more readable, even though it works without the change.

If others disagree I am happy to remove it.